File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webserver/service/message/digitaltwin/ReadEventsResponse.java
package org.opcfoundation.webserver.service.message.digitaltwin;
import org.opcfoundation.webserver.digitaltwin.event.EventData;

import java.util.ArrayList;
import java.util.List;

public class ReadEventsResponse {
    List<EventData> eventsData;
    boolean containsMoreData;

    public ReadEventsResponse()
    {
        eventsData = new ArrayList<>();
        containsMoreData = false;
    }

    public void addEventData(EventData eventData)
    {
        eventsData.add(eventData);
    }

    public List<EventData> getEventsData() {
        return eventsData;
    }

    public void setContainsMoreData(boolean containsMoreData) {
        this.containsMoreData = containsMoreData;
    }

    public boolean containsMoreData() {
        return containsMoreData;
    }
}

File: opcua-webservice-node/source/src/server/service/message/digitialtwin/ReadEventsResponse.ts
import { EventData } from "../../..";

export class ReadEventsResponse {
    private readonly _eventsData: Array<EventData>;
    private _containsMoreData: boolean;

    constructor() {
        this._eventsData = [];
        this._containsMoreData = false;
    }

    addEventData(eventData: EventData): void {
        this._eventsData.push(eventData);
    }

    get eventsData(): Array<EventData> {
        return this._eventsData;
    }

    set containsMoreData(containsMoreData: boolean) {
        this._containsMoreData = containsMoreData;
    }

    get containsMoreData(): boolean {
        return this._containsMoreData;
    }
}